controlplane/kubeadm/internal/controllers/helpers.go

```diff
@@ -332,5 +339,10 @@ func (r *KubeadmControlPlaneReconciler) generateMachine(ctx context.Context, kcp
 	if err := r.Client.Create(ctx, machine); err != nil {
 		return errors.Wrap(err, "failed to create machine")
 	}
+
+	// Remove the annotation tracking that a remediation is in progress (the remediation completed when
+	// the replacement machine have been created above).
+	delete(kcp.Annotations, controlplanev1.RemediationInProgressAnnotation)
```

Comment:
Do we have a reentrancy issue here?

I think if the controller fails after Machine creation and before the annotations is removed it won't be able to do any further remediations? (it always returns early in reconcileUnhealthyMachines)

Reply:
We'll add a log message. This should rarely happen and if it does this can be resolved manually.

We'll consider adding a condition in a follow-up issue